blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticIO.java

```diff
@@ -143,22 +140,25 @@ private IndexRequestBuilder createSingleRequest(Discovery md) throws IOException
     }
     
     public List<SearchResult> search(String tenant, String query) throws Exception {
-        // complain if someone is trying to search specifically on any tenant.
-        if (query.indexOf(tenantId.name()) >= 0) {
-            throw new Exception("Illegal query: " + query);
-        }
-        
```

Comment:
@tilogaat I am not sure what this line of code was doing. The original intent seemed to not allow a query which is cross-tenant, like tenant A cannot search for metrics belonging to tenant B. But all it was doing was searching to see if the string 'tenantId` does not exist in the query.
For what its worth, I checked whether cross-tenant discovery is possible, and it seems like its not. We index `tenantId` along with `metricName` and `unit` as terms, and while searching we pass the tenantId from the uri and use it as a `must` contain match, so its impossible to search for cross-tenant metrics.